x-pack/plugins/infra/public/components/log_stream/log_stream_embeddable.tsx

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { CoreStart } from 'kibana/public';
+
+import { I18nProvider } from '@kbn/i18n/react';
+import { EuiThemeProvider } from '../../../../observability/public';
+import { KibanaContextProvider } from '../../../../../../src/plugins/kibana_react/public';
+import { Query, TimeRange } from '../../../../../../src/plugins/data/public';
+import {
+  Embeddable,
+  EmbeddableInput,
+  IContainer,
+} from '../../../../../../src/plugins/embeddable/public';
+import { datemathToEpochMillis } from '../../utils/datemath';
+import { LazyLogStreamWrapper } from './lazy_log_stream_wrapper';
+
+export const LOG_STREAM_EMBEDDABLE = 'LOG_STREAM_EMBEDDABLE';
+
+export interface LogStreamEmbeddableInput extends EmbeddableInput {
+  timeRange: TimeRange;
+  query: Query;
+}
+
+export class LogStreamEmbeddable extends Embeddable<LogStreamEmbeddableInput> {
+  public readonly type = LOG_STREAM_EMBEDDABLE;
+  private node?: HTMLElement;
+
+  constructor(
+    private services: CoreStart,
+    initialInput: LogStreamEmbeddableInput,
+    parent?: IContainer
+  ) {
+    super(initialInput, {}, parent);
+  }
+
+  public render(node: HTMLElement) {
+    if (this.node) {
+      ReactDOM.unmountComponentAtNode(this.node);
+    }
+    this.node = node;
+
+    this.renderComponent();
+  }
+
+  public reload() {
+    this.renderComponent();
+  }
+
+  public destroy() {
+    super.destroy();
+    if (this.node) {
+      ReactDOM.unmountComponentAtNode(this.node);
+    }
+  }
+
+  private renderComponent() {
+    if (!this.node) {
+      return;
+    }
+
+    const startTimestamp = datemathToEpochMillis(this.input.timeRange.from);
+    const endTimestamp = datemathToEpochMillis(this.input.timeRange.to);
+    const query = processQuery(this.input.query);
+
+    if (!startTimestamp || !endTimestamp) {
+      return;
+    }
+
+    ReactDOM.render(
+      <I18nProvider>
+        <EuiThemeProvider>
+          <KibanaContextProvider services={this.services}>
+            <div style={{ width: '100%' }}>
+              <LazyLogStreamWrapper
+                startTimestamp={startTimestamp}
+                endTimestamp={endTimestamp}
+                height="100%"
+                query={query}
+              />
+            </div>
+          </KibanaContextProvider>
+        </EuiThemeProvider>
+      </I18nProvider>,
+      this.node
+    );
+  }
+}
+
+function processQuery(queryObj: Query): string | undefined {
+  if (!queryObj) {
+    return;
+  }
+
+  if (typeof queryObj === 'string') {
+    return queryObj;
+  }
+
+  if (queryObj.language === 'kuery') {
+    // `Query` is a funky type
+    return queryObj.query as string;
+  }
+
+  if (queryObj.language === 'lucene') {
```

Comment:
👍

Do you want to address that in a separate followup? Do we know if there's something in the `data` plugin that could help there?

Reply:
I wanna tackle this today in this PR. I haven't looked into it yet, but I think what makes the most sense is to allow the `<LogStream />` component to accept lucene queries, something like:

```
<LogStream
  query="whatever"
  queryLang="lucene"
/>
```

The `queryLang` will be optional and will default to `"KQL"`, so we don't break anything.

Do you think it makes sense?

Reply:
Cool, sounds good. There's a `Query` type provided by the data plugin, and it probably makes most sense to use that as our prop so we're following the conventions set there.

The ML anomaly swimlane embeddable uses that here: https://github.com/elastic/kibana/blob/master/x-pack/plugins/ml/public/embeddables/types.ts#L34, and the type looks like so: https://github.com/elastic/kibana/blob/master/src/plugins/data/common/query/types.ts#L12

So it's basically exactly what you've said, but it's nice to buy into the `data` stuff where we can.